modules/gapi/test/common/gapi_parsers_tests_common.hpp

```diff
@@ -0,0 +1,384 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+
+#ifndef OPENCV_GAPI_PARSERS_TESTS_COMMON_HPP
+#define OPENCV_GAPI_PARSERS_TESTS_COMMON_HPP
+
+#include "gapi_tests_common.hpp"
+#include "../../include/opencv2/gapi/core.hpp"
+
+namespace opencv_test
+{
+class ParsersTest
+{
+public:
+    cv::Mat generateSSDoutput(const cv::Size& in_sz)
+    {
+        constexpr int maxN = 200;
+        constexpr int objSize = 7;
+        std::vector<int> dims{ 1, 1, maxN, objSize };
+        cv::Mat mat(dims, CV_32FC1);
+        auto data = mat.ptr<float>();
+
+        for (int i = 0; i < maxN; ++i)
+        {
+            float* it = data + i * objSize;
+            auto ssdIt = generateItem(i, in_sz);
+            it[0] = ssdIt.image_id;
+            it[1] = ssdIt.label;
+            it[2] = ssdIt.confidence;
+            it[3] = ssdIt.rc_left;
+            it[4] = ssdIt.rc_top;
+            it[5] = ssdIt.rc_right;
+            it[6] = ssdIt.rc_bottom;
+        }
+
+        return mat;
+    }
+
+    cv::Mat generateYoloOutput(const int& num_classes)
+    {
+        std::vector<int> dims = { 1, 13, 13, (num_classes + 5) * 5 };
+        cv::Mat mat(dims, CV_32FC1);
+        auto data = mat.ptr<float>();
+
+        for (int i = 0; i < dims[0] * dims[1] * dims[2] * dims[3]; ++i)
+        {
+            data[i] = static_cast<float>(std::rand()) / RAND_MAX;
+        }
+        return mat;
+    }
+
+    void parseYoloRef(const cv::Mat& in_mat,
+                      const cv::Size& in_sz,
+                      const float& confidence_threshold,
+                      const float& nms_threshold,
+                      const int& num_classes,
+                      const cv::gapi::core::YoloAnchors& anchors,
+                      std::vector<cv::Rect>& out_boxes,
+                      std::vector<int>& out_labels)
+    {
+        auto detections = parse(in_mat.ptr<float>(),
+                                confidence_threshold,
+                                in_sz.height,
+                                in_sz.width,
+                                num_classes,
+                                anchors);
+
+        std::stable_sort(std::begin(detections), std::end(detections),
+                         [](const DetectionObject& a, const DetectionObject& b)
+                         {
+                             return a.confidence > b.confidence;
+                         });
```

Comment:
Is it necessary to make a new lambda when you've defined `operator </>` for `DetectionObject`?

Reply:
Yes. I removed unused operators.